glean-core/csharp/Glean/Metrics/CounterMetricType.cs

```diff
@@ -0,0 +1,146 @@
+﻿// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+using Mozilla.Glean.FFI;
+using System;
+
+namespace Mozilla.Glean.Private
+{
+    /// <summary>
+    /// This implements the developer facing API for recording counter metrics.
+    /// 
+    /// Instances of this class type are automatically generated by the parsers at build time,
+    /// allowing developers to record values that were previously registered in the metrics.yaml file.
+    ///
+    /// The internal constructor is only used by [LabeledMetricType] directly.
+    /// </summary>
+    public sealed class CounterMetricType : ILabeledSubmetricInterface
+    {
+        private bool disabled;
+        private string[] sendInPings;
+        private UInt64 handle;
+
+        /// <summary>
+        /// The public constructor used by automatically generated metrics.
+        /// </summary>
+        public CounterMetricType(
+            bool disabled,
+            string category,
+            Lifetime lifetime,
+            string name,
+            string[] sendInPings
+            ) : this(0, disabled, sendInPings)
+        {
+            handle = LibGleanFFI.glean_new_counter_metric(
+                        category: category,
+                        name: name,
+                        send_in_pings: sendInPings,
+                        send_in_pings_len: sendInPings.Length,
+                        lifetime: (int)lifetime,
+                        disabled: disabled);
+        }
+
+        internal CounterMetricType(
+            UInt64 handle,
+            bool disabled,
+            string[] sendInPings
+            )
+        {
+            this.disabled = disabled;
+            this.sendInPings = sendInPings;
+            this.handle = handle;
+        }
+
+        /// <summary>
+        /// Add to counter value.
+        /// </summary>
+        /// <param name="amount">
+        /// This is the amount to increment the counter by, defaulting to 1 if called without parameters.
+        /// </param>
+        public void Add(Int32 amount = 1)
+        {
+            if (disabled)
+            {
+                return;
+            }
+
+            Dispatchers.LaunchAPI(() => {
+                AddSync(amount);
+            });
+        }
+
+        /// <summary>
+        /// Add to counter value synchronously.
+        /// 
+        /// This is only to be used within the Glean SDK.
+        /// </summary>
+        /// <param name="amount">
```

Comment:
Minor nit about consistency of comment formatting, this `<param>` description starts with a capital letter and on a new line from the tag, where the other comments below start with a lower case letter on the same line.

Reply:
Good call. Let me make things more consistent.